examples/jest/__tests__/catAPI.spec.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const path = require('path')
+const { Pact, Matchers } = require('../../../dist/pact')
+const getMeCats = require('../').getMeCats
+
+describe("Cat's API", () => {
+  let url = 'http://localhost'
+
+  describe("another works", () => {
+    it('returns a successful body', async () => {
+      // add expectations
+      const EXPECTED_BODY = [
+        { cat: 2 }, { cat: 3 }
+      ]
+
+      const interaction = {
```

Comment:
I personally think this reads better in the `beforeEach()` block, because it's the Given part of the Given/When/Then (this also would match the other example tests in the Jest folder)

Reply:
This is probably a personal style thing. To explain what I meant, my pact tests tend to look like this:

```
describe("A request for foo", () => {
    describe("In state bar", ()=> {
        beforeEach(() => {/* setup interaction here*/}
        it("generates the right response", ()=> 
            api.foo().then(/* assertions go here */);
        )
    });
    describe("In state baz", ()=> {
        beforeEach(() => {/* setup interaction here*/}
        it("generates the right response", ()=> 
            api.foo().then(/* assertions go here */);
        )
    });
});
```

In this way, each test has a single corresponding `beforeEach()` statement. I like this, as it means that the `describe`s and the `beforeEach` are the "Given", the API call is the "When", and the `.then(/* assertions */)` is the "Then".  I find this very easy to read, because it doesn't include "given" sections within the `it()`. You may prefer a different style - there are many good ways to write a test.

I brought this up in the review not because I want you to use my style, but because I think it's valuable to have all the examples written in the same style, to minimise cognitive load on the reader.

A Pact test has a lot more going on than a standard unit test, so a nice clear separation of the different components is valuable to the new reader (and could easily be a companion to a documentation article).

I don't feel strongly about what style we use, as long as the style fits the values of being easy for a new reader (from the perspective of understanding Pact), and is a good example of idomatic code in the example test framework + language.

However, if we are going to change the style of the examples, I do feel strongly that we should change all of them.

Reply:
That makes sense though, thanks for explaining that.
I'd be happy to take this in another PR though and address all the examples so we can unblock this one.